snow/consensus/snowball/tree.go

```diff
@@ -557,14 +554,7 @@ func (b *binaryNode) RecordPoll(votes bag.Bag[ids.ID], reset bool) (node, bool)
 	}
 
 	if child := b.children[bit]; child != nil {
-		// The votes are filtered to ensure that they are votes that should
-		// count for the child
-		decidedPrefix := child.DecidedPrefix()
-		filteredVotes := prunedVotes.Filter(func(id ids.ID) bool {
-			return ids.EqualSubset(b.bit+1, decidedPrefix, b.preferences[bit], id)
-		})
-
-		newChild, _ := child.RecordPoll(filteredVotes, b.shouldReset[bit])
+		newChild, _ := child.RecordPoll(prunedVotes, b.shouldReset[bit])
 		if b.snow.Finalized() {
```

Comment:
It seems like the below `b.children[bit] = newChild` call is also useless code right?